Include goto-cl regression tests in test sequence [depends-on: #3674]

tautschnig opened the pull request:
Instead of special-casing the goto-cl folder via the CI configuration move all
target-platform specific logic into the (C)Makefile.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

tautschnig committed:
Include goto-cl regression tests in test sequence

Instead of special-casing the goto-cl folder via the CI configuration move all
target-platform specific logic into the (C)Makefile. Also make sure exclusion
patterns are used consistently.

tautschnig committed:
Get rid of Windows hacks

Fixes: #623
Fixes: #787

chrisr-diffblue reviewed (approved):
Assuming CI passes, this looks much cleaner - many thanks!

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: c9b1ae832f3db66b7ecee78f6b3274330d174bfc).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96163307

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: bca22076f6a96230130be329e5f192c80ea8fdcd).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96163777

tautschnig committed:
Move regression tests of the C front-end to a suitable folder, again

These tests aren't specific to building hybrid binaries, they just use parts of
C front-end that are GCC-specific. Thus place them in the ansi-c folder and use
the gcc-only tag.

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 5050350ab77a3b66a953f5212befcf37cf814436).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96168847

kroening commented on regression/cbmc/unsigned___int128/main.c:
```diff
@@ -1,3 +1,8 @@
+#if defined(_WIN32) && !defined(__CYGWIN__)
+void reduce()
```
I would check for the (absence of the) gcc define -- I believe __int128 is a gcc extension; so this isn't about Windows.

tautschnig committed:
C++ front-end: bool is also an integral type

Visual Studio uses enums with a bool subtype in header files.

tautschnig commented:
It seems we can get rid of the hack to remove directories in Windows builds once all the depended-on PRs are merged (with one yet to be created).

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: acc8d62dfb4736c7d0dd9b37d9dd66d301a981d3).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/97016238
